azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/OdataTypeDiscriminatorTypeResolver.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
```

Comment:
This is code generated?

Reply:
Not sure. It should. Current autogenerated code regard to "odata.type" is not correct.

Autogenerated code will generate "odata\\.type", which is invalid during deserialization.
However "odata.type" will generate wrong structure (nested type inside odata) during serialization.
So the solution is to use additional OdataTypeDiscriminatorTypeResolver to deserialze as "odata.type" and serialize as "odata\\.type", which is in previous code.

Ideally autogenator should create this class.

Reply:
Try fix it on https://github.com/Azure/autorest.java to ramp up with autorest

Reply:
Actually tried locally to fix ModelTemplate.cshtml. However adding one line and another file depending on whether these is a dot in Model.BasePolymorphicDiscriminator is too much for me at that time, so pending when I got some free time...